lib/spack/spack/directives.py

```diff
@@ -182,10 +192,28 @@ def _depends_on(pkg, spec, when=None):
         when = pkg.name
     when_spec = parse_anonymous_spec(when, pkg.name)
 
+    if deptypes is None:
+        # The default deptype is build and install because the common case is
+        # to build against a library which then turns into a runtime dependency
+        # due to the linker.
+        # XXX(deptype): Add DEP_RUN to this? It's an uncommon dependency type,
```

Comment:
I like adding `DEP_RUN`, both for backward compatibility and because a lot of libraries come with things that you may want to run with them (`libdwarf` comes with `dwarfdump`, etc. etc.).  We'll eventually use `DEP_RUN` to set up the module environment for each package, and in that case you probably do want the binaries (at least from `bin`) in the path.  I also think it's good for backward compatibility.

Can you change `DEP_INSTALL` to `DEP_LINK` or `DEP_LIB`?   I think that is closer to the intended purpose of the type.  `INSTALL` is a little ambiguous, since technically run dependencies need to be installed to run, and build dependencies need to be installed at build time.